### DISCUSSION TITLE
Issue streaming SB500 after upgrade to 4.0

### DISCUSSION THREAD

**Discussion by almoney** [GerryDazoo/Slinger#532] [Q&A]
I upgraded my slingers to 4.0 and have an issue streaming from my SB500 only everything else working fine.

This is in the log:

02/13/2023, 17:49:41.608   Streaming request from ('xxxx', 57770)  /ASling/Novi500 
STREAM=xxxx:57770:0
Starting remote stream 2

Slinginfo  "350/500/M1" 15 60 192.168.11.189 5221 3072 3 pw
Novi500 Starting Stream for  x.y.z.a:57770
Novi500 Opening Control stream 0x2000 0 0
Connecting... ('192.168.11.189', 5221) Control
Novi500 Selecting Video Source 1
Novi500 Keycodes= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 23, 24, 25, 26, 27, 28, 29, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 62, 63, 64]
VideoParameters: Resolution= 15 FrameRate= 60 VideoBandwidth= 8000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 96
Connecting... ('192.168.11.189', 5221) Stream
**Traceback (most recent call last):
  File "/home/slingbox/slingbox_server.py", line 836, in streamer
    s_ctl, stream  = start_slingbox_session(streams)
  File "/home/slingbox/slingbox_server.py", line 428, in start_slingbox_session
    first_buffer = bytearray(stream.recv(pksize, socket.MSG_PEEK))
socket.timeout: timed out
Novi500 Badness starting slingbox session  timed out None**
Streamer:  Novi500 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

Config for this box:

[Novi500]
sbtype="350/500/M1"
password=removed
ipaddress=192.168.11.189
port=5221
Resolution=15
FrameRate=60
AudioBitRate=96
VideoBandwidth=8000
VideoSmoothness=63
VideoSource=1
maxstreams=3

Any thoughts?
(works fine after I roll back.)

**Comment by GerryDazoo** [comment]
I don't see anything obviously wrong. Are you running the python or Windows exe? I've started up 100's of streams to my 500 with this code.  Please give the 500 a quick power cycle. Your really need to post ALL the server output. But it looks like the 500 is going radio silent after the video configuration command. 
Just for fun can you try a different resolution (12).

**Reply by blasco403** [reply]
I have upgraded to 4.0 and have a 500 and have had no issues. I am using resolution 12.

**Comment by gndenton** [comment]
I am experiencing a similar issue - here's the log output:

Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Building page for Slingbox
Reading Custom Remote definition from remote.txt
Starting Streamer Thread for Slingbox
Streamer Running: starting up on  port 8080  108388608 
config.ini SLINGBOX 8080 8080 8388608
8080 Checking for slingbox at ('192.168.1.26', 5301) 0
8080 ('192.168.1.26', 5301) OK
8080 Using slingbox at  ('192.168.1.26', 5301)
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
 * Serving Flask app 'slingbox_server'
 * Debug mode: off
Broadcast Responder Running
Server IP address is 192.168.1.12
06/15/2023, 15:03:46.449   Streaming request from ('192.168.1.127', 51570)  /slingbox 
STREAM=192.168.1.127:51570:0

Slinginfo  "350/500/M1/M2" 11 30 192.168.1.26 5301 3072 10 "password"
8080 Starting Stream for  192.168.1.127:51708
8080 Opening Control stream 0x2000 0 0
Connecting... ('192.168.1.26', 5301) Control
8080 Selecting Video Source 1
8080 Keycodes= [1, 2, 3, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 24, 27, 28, 29, 30, 31, 33, 35, 36, 38, 39, 40, 41, 42, 46, 62, 63, 64]
VideoParameters: Resolution= 11 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.26', 5301) Stream
8080 Stream started at 06/15/2023, 15:07:06.951  745 0
.06/15/2023, 15:07:25.866  8080 Stream Terminated for  192.168.1.127:51708
8080 Shutting down connections
8080 Logging Out
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
06/15/2023, 15:08:41.063   Streaming request from ('192.168.1.127', 51785)  /slingbox 
STREAM=192.168.1.127:51785:0


I have tried differing resolutions, with the same result.  The only thing displayed on the VLC window is "8080 - 8080" at the bottom center of the screen.  It is displayed for about 2 seconds and is then gone.

**Reply by GerryDazoo** [reply]
Try a different media player. I've given up on sorting out all the issues with VLC.
If this one  https://github.com/GerryDazoo/SlingerPlayer doesn't work then we can see what might be going wrong.

**Reply by skijohn877** [reply]
Yeah, VLC is a dog, I have had issues with my 500 with it, yet works with my 350's with no issues. SlingerPlayer works great with both the 500 and 350.

**Comment by gndenton** [comment]
Thanks for the response.  I downloaded and installed the SlingerPlayer.  I’m not 100% sure of the root of the problem, but both VLC and SlingerPlayer provided audio, but no video.  Perhaps a config.ini setting?  I’ll keep digging on this one.  BTW — nice addition with the SlingerPlayer (iOS soon?).

Gary N. Denton
Mobile: +1 (317) 430-3051
Email: ***@***.***

________________________________

**Reply by skijohn877** [reply]
Check to make sure you have the correct video source in config, maybe try the component setting

**Reply by GerryDazoo** [reply]
Yes, that's the most plausible problem. You've selected a video source that has no input.
Sorry, no iOS. I have no way to test it.

**Comment by gndenton** [comment]
Thanks again.  The interesting part is two-fold:

  1.  I have video (component in and component out through the Slingbox M1) to a TV.  Looks and sounds acceptable
  2.  I had audio only from both SlingerPlayer and VLC


As I stated, I’ll keep digging.  It has to be something on my setup.

iOS — understand.  I appreciate all you have done to save the Slingbox world!


Gary N. Denton
Mobile: +1 (317) 430-3051
Email: ***@***.******@***.***>

**Comment by gndenton** [comment]
Gerry – issue solved for me. After further debug, I found a malfunctioning HDMI-->Component converter (input to Slingbox M1).  For whatever reason, the pass-thru to the component output on the M1 showed no issues.

Thanks again for the assistance.


Gary N. Denton
Mobile: +1 (317) 430-3051
Email: ***@***.******@***.***>

From: Gary Denton
Sent: Saturday, June 17, 2023 11:49 AM
To: GerryDazoo/Slinger ***@***.***>; GerryDazoo/Slinger ***@***.***>
Cc: Comment ***@***.***>
Subject: RE: [GerryDazoo/Slinger] Issue streaming SB500 after upgrade to 4.0 (Discussion #532)

Thanks again.  The interesting part is two-fold:

  1.  I have video (component in and component out through the Slingbox M1) to a TV.  Looks and sounds acceptable
  2.  I had audio only from both SlingerPlayer and VLC


As I stated, I’ll keep digging.  It has to be something on my setup.

iOS — understand.  I appreciate all you have done to save the Slingbox world!


Gary N. Denton
Mobile: +1 (317) 430-3051
Email: ***@***.******@***.***>
